covfee/loader.py

```diff
@@ -30,82 +32,109 @@ def cli_create_tables():
 class Loader:
     """Translates between different covfee file formats"""
 
-    def __init__(self, project_spec_file=None):
+    _project_spec_file: Path
+    _project_spec_from_python_file: bool
+
+    def __init__(self, project_spec_file: str):
         if not os.path.exists(project_spec_file):
             raise FileNotFoundError("covfee file not found.")
 
-        self.project_spec_file = Path(project_spec_file)
-        self.file_extension = self.project_spec_file.suffix
-        if self.file_extension not in [".py", ".json"]:
-            raise ValueError(f"Unsupported file extension {self.file_extension}")
+        self._project_spec_file = Path(project_spec_file)
+        self._project_spec_from_python_file = self._project_spec_file.suffix == ".py"
+
+        if (
+            not self._project_spec_from_python_file
+            and self._project_spec_file.suffix != ".json"
+        ):
+            raise ValueError(
+                f"Unsupported covfee project specification file type {project_spec_file}"
+            )
+
+    def load_project_spec_file_and_parse_as_covfee_app(
+        self, with_spinner=False
+    ) -> CovfeeApp:
+        covfee_app: CovfeeApp
+        if self._project_spec_from_python_file:
+            covfee_app = self._load_covfee_app_from_python_module()
+        else:
+            schema = Schemata()
+            if not schema.exists():
+                schema.make()
```

Comment:
I know this was copy pasted, but why are we running schemata make if it is done automatically here?

Reply:
I honestly have no clue. Indeed, this is just a copy/paste to make the changes progressively.